src/nnet/nnet-parametric-relu.h

```diff
@@ -0,0 +1,266 @@
+  }
+
+  void PropagateFnc(const CuMatrixBase<BaseFloat> &in,
+                    CuMatrixBase<BaseFloat> *out) {
+    // Multiply activations by ReLU scalars (max_alpha, min_beta):
+    // // out = in * (in >= 0.0 ? in * max_alpha : in * min_beta)
+    out->ParametricRelu(in, alpha_, beta_);
+  }
+
+  void BackpropagateFnc(const CuMatrixBase<BaseFloat> &in,
+                        const CuMatrixBase<BaseFloat> &out,
+                        const CuMatrixBase<BaseFloat> &out_diff,
+                        CuMatrixBase<BaseFloat> *in_diff) {
+    // multiply error derivative by activations (alpha, beta)
+    // ((dE/da)*w) === out_diff, f(y) == out,
+    //  (out > 0 ) ? out_diff * alpha : out_diff * beta
+    in_diff->DiffParametricRelu(out, out_diff, alpha_, beta_);
+  }
+
+
+  void Update(const CuMatrixBase<BaseFloat> &input,
+              const CuMatrixBase<BaseFloat> &diff) {
+    // we use following hyperparameters from the option class
+    const BaseFloat alr = opts_.learn_rate * bias_learn_rate_coef_;
+    const BaseFloat blr = opts_.learn_rate * learn_rate_coef_;
+    const BaseFloat mmt = opts_.momentum;
+    if (clip_gradient_ > 0.0) {  // gradient clipping
+      alpha_corr_.ApplyFloor(-clip_gradient_);
+      beta_corr_.ApplyFloor(-clip_gradient_);
+      alpha_corr_.ApplyCeiling(clip_gradient_);
+      beta_corr_.ApplyCeiling(clip_gradient_);
+    }
+    // compute gradient (incl. momentum)
+    if (!fix_alpha_) {  // the alpha parameter is learnable
+       in_alpha_.Resize(input.NumRows(), input.NumCols());
+       in_alpha_.CopyFromMat(input);
+       in_alpha_.ApplyFloor(0.0);
+       in_alpha_.MulElements(diff);
+       alpha_corr_.AddRowSumMat(1.0, in_alpha_, mmt);
+       alpha_.AddVec(-alr, alpha_corr_);
+    }
+    if (!fix_beta_) {  // the beta parameter is learnable
+       in_beta_.Resize(input.NumRows(), input.NumCols());
+       in_beta_.CopyFromMat(input);
+       in_beta_.ApplyCeiling(0.0);
+       in_beta_.MulElements(diff);
+       beta_corr_.AddRowSumMat(1.0, in_beta_, mmt);
+       beta_.AddVec(-blr, beta_corr_);
+    }
+  }
+
+  /// Accessors to the component parameters,
+  const CuVectorBase<BaseFloat>& GetAlpha() const { return alpha_; }
+
+  void SetAlpha(const CuVectorBase<BaseFloat>& alpha) {
+    KALDI_ASSERT(alpha.Dim() == alpha_.Dim());
+    alpha_.CopyFromVec(alpha);
+  }
+  const CuVectorBase<BaseFloat>& GetBeta() const { return beta_; }
+
+  void SetBeta(const CuVectorBase<BaseFloat>& beta) {
+    KALDI_ASSERT(beta.Dim() == beta_.Dim());
+    beta_.CopyFromVec(beta);
+  }
+
+ private:
+  CuVector<BaseFloat> alpha_; /// < Vector of 'alphas', one value per neuron.
+  CuVector<BaseFloat> beta_; /// < Vector of 'betas', one value per neuron.
+
+  CuVector<BaseFloat> alpha_corr_; /// < Vector of 'alpha' updates.
+  CuVector<BaseFloat> beta_corr_; /// < Vector of 'beta' updates.
+
+  CuMatrix<BaseFloat> in_alpha_;
+  CuMatrix<BaseFloat> in_beta_;
+
+  BaseFloat clip_gradient_;
```

Comment:
It looks like the variable 'clip_gradient_' is unused... It would be better to either use it or remove it...